tests/PHPStan/Analyser/nsrt/bug-5017.php

```diff
@@ -12,10 +12,10 @@ public function doFoo()
 		$items = [0, 1, 2, 3, 4];
 
 		while ($items) {
-			assertType('non-empty-array<0|1|2|3|4, 0|1|2|3|4>', $items);
+			assertType('non-empty-list<int<min, 4>>', $items);
 			$batch = array_splice($items, 0, 2);
-			assertType('array<0|1|2|3|4, 0|1|2|3|4>', $items);
-			assertType('non-empty-list<0|1|2|3|4>', $batch);
+			assertType('list<int<min, 4>>', $items);
+			assertType('non-empty-list<int<min, 4>>', $batch);
```

Comment:
the changes here confused me briefly, but it seems to be coming from loop generalization and there's not much I can do about it